src/find/matchers/size.rs

```diff
@@ -0,0 +1,174 @@
+        false
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use find::matchers::{ComparableValue, Matcher};
+    use find::matchers::tests::get_dir_entry_for;
+    use find::tests::FakeDependencies;
+    use super::*;
+    // need to explicitly use non-pub members
+    use super::{byte_size_to_unit_size, Unit};
+
+    #[test]
+    fn test_byte_size_to_unit_size() {
+        assert_eq!(byte_size_to_unit_size(Unit::KiloByte, 0), 0);
+        assert_eq!(byte_size_to_unit_size(Unit::KiloByte, 1), 1);
+        assert_eq!(byte_size_to_unit_size(Unit::KiloByte, 1024), 1);
+        assert_eq!(byte_size_to_unit_size(Unit::KiloByte, 1025), 2);
+        assert_eq!(byte_size_to_unit_size(Unit::Byte, 1025), 1025);
+        assert_eq!(byte_size_to_unit_size(Unit::TwoByteWord, 1025), 513);
+        assert_eq!(byte_size_to_unit_size(Unit::Block, 1025), 3);
+        assert_eq!(byte_size_to_unit_size(Unit::KiloByte, 1025), 2);
+        assert_eq!(byte_size_to_unit_size(Unit::MegaByte, 1024 * 1024 + 1), 2);
+        assert_eq!(byte_size_to_unit_size(Unit::GigaByte, 1024 * 1024 * 1024 + 1),
+                   2);
+    }
+
+    #[test]
+    fn unit_from_string() {
+        assert_eq!(byte_size_to_unit_size(Unit::from_string("c").unwrap(), 2),
+                   2);
+        assert_eq!(byte_size_to_unit_size(Unit::from_string("w").unwrap(), 3),
+                   2);
+        assert_eq!(byte_size_to_unit_size(Unit::from_string("b").unwrap(), 513),
+                   2);
+        assert_eq!(byte_size_to_unit_size(Unit::from_string("").unwrap(), 513),
+                   2);
+        assert_eq!(byte_size_to_unit_size(Unit::from_string("k").unwrap(), 1025),
+                   2);
+        assert_eq!(byte_size_to_unit_size(Unit::from_string("M").unwrap(), 1024 * 1024 + 1),
+                   2);
+        assert_eq!(byte_size_to_unit_size(Unit::from_string("G").unwrap(), 2024 * 1024 * 1024 + 1),
+                   2);
+    }
+
+    #[test]
+    fn size_matcher_bad_unit() {
+        if let Err(e) = SizeMatcher::new(ComparableValue::EqualTo(2), "xyz") {
+            assert!(e.description().contains("Invalid suffix") && e.description().contains("xyz"),
+                    "bad description: {}",
+                    e);
+        } else {
+            panic!("parsing a unit string should fail");
+        }
+    }
+
+    #[test]
+    fn size_matcher() {
+        let file_info = get_dir_entry_for("./test_data/size", "512bytes");
+
+        let equal_to_2_blocks = SizeMatcher::new(ComparableValue::EqualTo(2), "b").unwrap();
+        let equal_to_1_blocks = SizeMatcher::new(ComparableValue::EqualTo(1), "b").unwrap();
+        let deps = FakeDependencies::new();
+
+        assert!(!equal_to_2_blocks.matches(&file_info, &mut deps.new_matcher_io()),
+                "512-byte file should not match size of 2 blocks");
+        assert!(equal_to_1_blocks.matches(&file_info, &mut deps.new_matcher_io()),
+                "512-byte file should match size of 1 block");
+    }
+
+
```

Comment:
My old nemesis: vertical whitespace :-(

Alas rustfmt doesn't help here (My eclipse workspace runs it at every save)